x-pack/plugins/triggers_actions_ui/public/application/lib/rule_api/load_execution_log_aggregations.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/* eslint-disable @typescript-eslint/naming-convention */
+
+import { HttpSetup } from 'kibana/public';
+import { INTERNAL_BASE_ALERTING_API_PATH } from '../../constants';
+import { IExecutionLogResult, IExecutionLog } from '../../../../../alerting/common';
+import { AsApiContract, RewriteRequestCase } from '../../../../../actions/common';
+
+const getRenamedLog = (data: IExecutionLog) => {
+  const {
+    duration_ms,
+    total_search_duration_ms,
+    es_search_duration_ms,
+    schedule_delay_ms,
+    ...rest
+  } = data;
+
+  return {
+    execution_duration: data.duration_ms,
+    total_search_duration: data.total_search_duration_ms,
+    es_search_duration: data.es_search_duration_ms,
+    schedule_delay: data.schedule_delay_ms,
+    ...rest,
+  };
+};
+
+const rewriteBodyRes: RewriteRequestCase<IExecutionLogResult> = ({ data, total }: any) => ({
+  data: data.map((log: IExecutionLog) => getRenamedLog(log)),
+  total,
+});
+
+const getFilter = (filter: string[] | undefined) => {
+  if (!filter || !filter.length) {
+    return;
+  }
+  return filter.join(' OR ');
+};
+
+export type SortFields =
+  | 'timestamp'
+  | 'execution_duration'
+  | 'total_search_duration'
+  | 'es_search_duration'
+  | 'schedule_delay'
+  | 'num_triggered_actions';
+
+export type SortOrder = 'asc' | 'desc';
```

Comment:
We shouldn't need to explicitly define this - maybe we can use `estypes.SortOrder`?